plugins/woocommerce/includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -392,4 +487,115 @@ protected function prepare_column_headers() {
 			$this->get_sortable_columns(),
 		);
 	}
+
+	/**
+	 * Convert an array or object into a multidimensional array.
+	 *
+	 * @param array|object $input Array or object to be processed.
+	 *
+	 * @return array
+	 */
+	public function to_multidimensional_array( $input ) {
+		$out   = array();
+		$input = is_object( $input ) ? (array) $input : $input;
+
+		if ( is_array( $input ) ) {
+
+			// Run function on each item in the array.
+			$out = array_map(
+				function( $value ) {
+
+					if ( is_array( $value ) || is_object( $value ) ) {
+						return $this->to_multidimensional_array( $value );
+					}
+
+					return $value;
+				},
+				$input
+			);
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Convert an array to a string with tabbed indentaion.
+	 *
+	 * @param array|object $input   An object or array.
+	 * @param array        $exclude An array of strings of params/keys to exclude.
+	 * @param int          $depth   The current depth of the walker.
+	 *
+	 * @return string
+	 */
+	public function printable_array( $input, $exclude = array(), $depth = 0 ) {
+		$out   = '';
+		$input = is_object( $input ) ? (array) $input : $input;
+		if ( is_array( $input ) ) {
+
+			foreach ( $input as $key => $val ) {
+
+				$key = strip_tags( (string) $key );
+
+				if ( in_array( (string) $key, $exclude ) ) {
+					continue;
+				}
+
+				$tabs = '';
+				for ( $i = 0; $i < $depth; $i++ ) {
+					$tabs .= '  ';
+				}
+
+				$out .= $tabs . '[' . $key . '] => ';
+
+				if ( is_array( $val ) || is_object( $val ) ) {
+					$new_depth = $depth + 1;
+					$out .= '[&#010;' . $this->printable_array( $val, $exclude, $new_depth ) . $tabs . ']';
+				} else {
+					$out .= "'" . $val . "'";
+				}
+				$out .= '&#013;&#010;';
+			}
+		}
+		$out .= '';
+		return $out;
+	}
+
+	/**
+	 * Outputs JS to control display of "Error Details".
+	 *
+	 * Not in a separate file as this is a single-use script.
+	 */
+	public function toggle_details_script() {
+		?>
+		<script type="text/javascript" id="wc-log-details">
+			( function( $ ) {
+				$( '#the-list' ).on( 'click', '.details-button', function( e ) {
+					e.preventDefault();
+
+					var button = $( this ),
+					logId = button.attr( 'data-details-button' ),
+					box = $( "[data-details-box='" + logId + "']" ),
+					defaultText = '<?php esc_html_e( 'View Details', 'woocommerce' ); ?>',
+					hideText = '<?php esc_html_e( 'Hide Details', 'woocommerce' ); ?>';
```

Comment:
```suggestion
					let $button     = $( this ),
					let logId       = button.attr( 'data-details-button' ),
					let $box        = $( "[data-details-box='" + logId + "']" ),
					let defaultText = '<?php esc_html_e( 'View Details', 'woocommerce' ); ?>',
					let hideText    = '<?php esc_html_e( 'Hide Details', 'woocommerce' ); ?>';
```

The `$` prefix for jQuery objects is a useful convention (and we'd also update any subsequent references).